langgenius/dify
# feat: Add program_name attribute to TiDB connection

# Description

Add program_name parameter in SQLAlchemy's connection string. It is empty when not specified otherwise.

Fixes #5500

## Type of Change

Please delete options that are not relevant.


# How Has This Been Tested?

1. Configuring docker-compose.yaml to use tidb_vector as vector store like this.
```yaml
version: '3'
services:
  api:
    environment:
      VECTOR_STORE: tidb_vector
      TIDB_VECTOR_HOST: tidb
      TIDB_VECTOR_PORT: 4000
      TIDB_VECTOR_USER: xxx.root
      TIDB_VECTOR_PASSWORD: xxxxxx
      TIDB_VECTOR_DATABASE: dify
  worker:
    environment:
      VECTOR_STORE: tidb_vector
      TIDB_VECTOR_HOST: tidb
      TIDB_VECTOR_PORT: 4000
      TIDB_VECTOR_USER: xxx.root
      TIDB_VECTOR_PASSWORD: xxxxxx
      TIDB_VECTOR_DATABASE: dify

```
2. Start a local deployment of dify with docker compose and patch the docker-worker-1 and docker-api-1 containers with the updated `tidb_vector.py`. 
3. Verify the new connection string works

# Suggested Checklist:
